test/integration/test_use_proxy.py

```diff
@@ -0,0 +1,75 @@
+"""Integration test for agent-daemon communication through MITM proxy."""
+import asyncio
+from functools import partial
+from pathlib import Path
+
+import pytest
+
+
+from goth.configuration import load_yaml
+from goth.runner import Runner
+from goth.runner.probe import RequestorProbe
+
+from goth.assertions import EventStream
+from goth.api_monitor.api_events import APIEvent, APIRequest
+
+
+async def api_call_made(container_name: str, stream: EventStream[APIEvent]) -> bool:
+    """Assert that an API call to `container_name` is made."""
+
+    async for event in stream:
+        if isinstance(event, APIRequest) and event.callee == f"{container_name}:daemon":
+            return True
+    raise AssertionError(f"No API call to {container_name} registered by proxy")
+
+
+async def no_api_call_made(container_name: str, stream: EventStream[APIEvent]) -> bool:
+    """Assert that no API call to `container_name` is made."""
+
+    try:
+        await api_call_made(container_name, stream)
+    except AssertionError:
+        return True
+    raise AssertionError(f"API call to {container_name} registered by proxy")
+
+
+@pytest.mark.asyncio
+async def test_use_proxy(default_goth_config: Path, log_dir: Path) -> None:
+    """Test if runner correctly sets up agent-deamon communication through proxy."""
```

Comment:
```suggestion
    """Test if runner correctly sets up agent-daemon communication through proxy."""
```

Reply:
Damn it, again! This happens so often to me it's starting to get embarassing...